src/main/webapp/app/text-editor/text-editor.component.ts

```diff
@@ -142,6 +144,24 @@ export class TextEditorComponent implements OnInit {
         }
 
         this.submission.text = this.answer;
+        const languageProbabilities = franc.all(this.answer);
+
+        for (const languageProbability of languageProbabilities) {
```

Comment:
Should we extract this into its own Service offering a method to return a language (or `null`) for a given string?
Also, consider using a `switch` Statement.

Reply:
I have to` break;` the outer `for` loop in the if clause. If I would use`break;` in a `switch` statement this would refer to the `switch` not to the`for` loop

Reply:
Maybe use the [`only` option](https://github.com/wooorm/franc#only) and get rid of the for loop?

Reply:
That's indeed a much better solution